Cython/Compiler/TypeSlots.py

```diff
@@ -503,6 +503,17 @@ def spec_value(self, scope):
         return self.slot_code(scope)
 
 
+class DelSlot(SyntheticSlot):
+    # For __del__ if defined
+    def __init__(self, slot_name, user_methods, default_value, **kwargs):
+        super(DelSlot, self).__init__(slot_name, user_methods, default_value, **kwargs)
+
+    def slot_code(self, scope):
+        if not scope.lookup_here("__del__"):
+            return 0
+        return InternalMethodSlot.slot_code(self, scope)
```

Comment:
I'm not 100% sure what's going on with PyPy, but I think the fact this slot is even set is breaking it. I'd change this line to

```
return """
#if CYTHON_USE_TP_FINALIZE
%s
#else
    NULL
#endif
""" % InternalMethodSlot.slot_code(self, scope)
```

and that way avoid setting `tp_finalize` unless we plan to use it